src/main/java/edu/nicholaidudakiwwarrick/advancedjava/app/StockQuoteApplication.java

```diff
@@ -0,0 +1,180 @@
+package edu.nicholaidudakiwwarrick.advancedjava.app;
+
+import edu.nicholaidudakiwwarrick.advancedjava.model.StockQuote;
+import edu.nicholaidudakiwwarrick.advancedjava.services.ServiceFactory;
+import edu.nicholaidudakiwwarrick.advancedjava.services.ServiceType;
+import edu.nicholaidudakiwwarrick.advancedjava.services.StockService;
+import edu.nicholaidudakiwwarrick.advancedjava.services.StockServiceException;
+import edu.nicholaidudakiwwarrick.advancedjava.util.IntervalEnum;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormatter;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * Basic class with a Main method to create StockService instances
+ */
+public class StockQuoteApplication {
+
+    @Option(name="-symbol", usage="stock symbol to look up")
+    private String symbol;
+
+    @Option(name="-start", usage="start date of the stock lookup")
+    private String startDate;
+
+    @Option(name="-end", usage="end date of the stock lookup")
+    private String endDate;
+
+    @Option(name="-interval", usage="time interval between the start and end dates")
+    private String interval;
+
+    /**
+     * @param args an array that contains elements:
+     *             <code>String</code> representation of stock symbol, start date, and end date
+     */
+    public static void main(@NotNull String[] args) {
```

Comment:
break up this up into discrete methods (where each method does one thing) - it is way too long.